### ISSUE TITLE
Ignore UTF-8 byte order mark

### ISSUE THREAD

**Issue by jclc** [munificent/wren#520]
Apparently the Wren compiler doesn't like when the UTF-8 byte order mark is present.
```
$ wren test.wren 
[main line 1] Error: Invalid byte 0xef.
[main line 1] Error: Invalid byte 0xbb.
[main line 1] Error: Invalid byte 0xbf.
```
The Unicode standard mandates that a UTF-8 encoded file may have this byte order mark. Although discouraged by pretty much everyone, Microsoft insists on using it on all UTF-8 encoded text files. If the byte order mark is missing, Notepad, for instance, will try to open it using another encoding.

The Wren compiler should just skip these bytes if they are the first 3 bytes in a file.

**Comment by bjorn** [comment]
Maybe this helps (I had to deal with this problem in another project when loading Lua scripts, cause Lua tripped over this as well):

```c
static char *skipPotentialBom(char *text)
{
    // Based on the C version of bomstrip
    const char * const utf8Bom = "\xef\xbb\xbf";
    const int bomLength = strlen(utf8Bom);
    return (strncmp(text, utf8Bom, bomLength) == 0) ? text + bomLength : text;
}
```

**Comment by jclc** [comment]
I fixed this in wren_compiler.c by adding an offset, but the tests fail since I forked the master branch which is also failing. Not sure if I should make a pull request...

**Comment by munificent** [comment]
Fixed by #525. Thanks, @jclc!
